pkgs/tools/admin/yandexcloud-cli/default.nix

```
@@ -0,0 +1,121 @@
+{ lib
+, stdenv
+, fetchurl
+, makeBinaryWrapper
+, installShellFiles
+, testers
+  # update script
+, writeShellScript
+, curl
+, common-updater-scripts
+}:
+let
+  version = "0.107.0";
+  hashes = {
+    # Linux
+    "aarch64-linux" = "sha256-KCvztarM9Y3ul+dQD8yscZ9or79tChi6SOvv7SQc0w8=";
+    "x86_64-linux" = "sha256-DGueeBc2tFm94cI+TlbjFVxCCrGljaJGIFInXt9Dnp0=";
+    "i686-linux" = "sha256-5P7zu1P6SBazJV5DVEVim6mqhFOW0FNXy3MUhjqZ5eA=";
+    # Darwin
+    "aarch64-darwin" = "sha256-NYj/9rhMglDtgnQVKRAL96q0SXlDzd+3km5G2mpctIE=";
+    "x86_64-darwin" = "sha256-J7rAxUEw50bsWb9lCxKWraH+d1WnH/fWcsOQ1RkuJ0s=";
+    # Windows
+    "x86_64-windows" = "sha256-aT+FYc3FD8L6KSD3pNvvSbUkNCAyofkfjGBmGN96OS0=";
+    "i686-windows" = "sha256-wqSUspHcMqFKaXILK7PdnnnoWSLycL+4HDXXya2iTCI=";
+  };
+
+  storageURL = "https://storage.yandexcloud.net/yandexcloud-yc/release";
+
+  fetchSource = system: hash:
+    let
+      throwSystem = throw "Unsupported system: ${system}";
+      platform = lib.systems.elaborate system;
+      arch =
+        if platform.isAarch64 then "arm64"
+        else if platform.isx86_64 then "amd64"
+        else if platform.isx86_32 then "386"
+        else throwSystem;
+      os =
+        if platform.isLinux then "linux"
+        else if platform.isDarwin then "darwin"
+        else if platform.isWindows then "windows"
+        else throwSystem;
+      bin = "yc${lib.optionalString platform.isWindows ".exe"}";
+    in
+    fetchurl {
+      url = "${storageURL}/${version}/${os}/${arch}/${bin}";
+      inherit hash;
+      executable = true;
+    };
+in
+stdenv.mkDerivation (finalAttrs:
+let
+  # For generating shell completion even when cross-compiling. Note that
+  # binaries are statically linked so we can execute them without patching.
+  nativeBin = finalAttrs.passthru.sources.${stdenv.buildPlatform.system} or null;
```

This should be using canExecute

---

So iterate over sources until a system that build platform can execute is found? Probably in the order of buildPlatform.system → hostPlatform.system → others.

---

I haven't seen this before in the code and would like to avoid the special exception here

```suggestion
  nativeBin = stdenv.buildPlatform.system} or null;
```